Fix the larva hive assignment to forsaken to only occur after the shuttle has been launched via hijack

rattybag opened the pull request:
I've become aware of the side effects the merge of my PR https://github.com/cmss13-devs/cmss13/pull/9211 have caused, essentially setting groundside xenos to forsaken at the wrong time.

I believe the root cause of this issue is because the check I introduced uisng the new hijack state `HIJACK_OBJECTIVES_SHIP_INBOUND` was happening when the shuttle was **called**, rather than when it had **launched**.

This PR adds a new hijack state, `HIJACK_OBJECTIVES_SHIP_IN_FLIGHT`, which is set when the shuttle is in flight after being triggered by hijack. 

I have tested that this correctly sets groundside larva / embryos to forsaken only when the shuttle is **launched**, rather than **called**, and does not have the side effect of changing existing xenos to forsaken when the shuttle is called.

I have reverted the changes made to the [for loop](https://github.com/Drulikar/cmss13/compare/forsaken_fixes...rattybag:cmss13:forsaken_fixes?expand=1#diff-3155cd7838a11267b5878f4a7af5eb8ab0b88960f6cc8bce8592181400bb6181L806-L816) to demonstrate that this is working. (Probably not necessary)

Unsure how you want to proceed based on this, but I do feel some ownership of this change since I've been working on it for a few months. 

Perhaps we can implement some combination of these changes, as I think with the reversion of the changes I introduced [here](https://github.com/Drulikar/cmss13/compare/forsaken_fixes...rattybag:cmss13:forsaken_fixes?expand=1#diff-5196fcddc2536ceb646edfbd84f21b37ab6ba05ec4a130f5bd3d0de66a46647bR176-R179) and [here](https://github.com/Drulikar/cmss13/compare/forsaken_fixes...rattybag:cmss13:forsaken_fixes?expand=1#diff-5196fcddc2536ceb646edfbd84f21b37ab6ba05ec4a130f5bd3d0de66a46647bR176-R179), http://github.com/cmss13-devs/cmss13/issues/9008 may not be fixed. 

I also want to apologise for breaking things in this unintended way, I did try to do comprehensive testing, but ultimately they were just smoke tests which is why this edge case eluded me.

If you want to merge this PR I can add back the changes to [the for loop](https://github.com/Drulikar/cmss13/compare/forsaken_fixes...rattybag:cmss13:forsaken_fixes?expand=1#diff-3155cd7838a11267b5878f4a7af5eb8ab0b88960f6cc8bce8592181400bb6181L806-L816) so that these changes remain in:

- Now abandon on hijack occurs when the shuttle actually launches
- facehugger masks on a person (e.g. hugging someone) will be converted

(I'm just a little unsure as to whether these will fix the underlying issue with stranded planetside xenos from bursts not being set to forsaken - I don't think these changes address that edge case)

<img width="629" height="398" alt="image" src="https://github.com/user-attachments/assets/367a9505-f8f6-4df2-bc36-10d3b7cc1fd4" />

Video example:
https://www.youtube.com/watch?v=dzB30VTasmI

Drulikar commented:
As discussed in DMs I addressed all issues I wanted addressed in my PR. I do not want these changes.

rattybag commented:
> As discussed in DMs I addressed all issues I wanted addressed in my PR. I do not want these changes.

All good - I would have made this a draft PR as I made this more to demonstrate my thought process rather than to have this merged, but I couldn't see a way to make it a draft
